cmd/argo/commands/clustertemplate/update.go

```diff
@@ -0,0 +1,75 @@
+package clustertemplate
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/argoproj/argo-workflows/v3/cmd/argo/commands/client"
+	"github.com/argoproj/argo-workflows/v3/pkg/apiclient/clusterworkflowtemplate"
+)
+
+type cliUpdateOpts struct {
+	output string // --output
+	strict bool   // --strict
+}
+
+func NewUpdateCommand() *cobra.Command {
+	var cliUpdateOpts cliUpdateOpts
+	command := &cobra.Command{
+		Use:   "update FILE1 FILE2...",
+		Short: "update a cluster workflow template",
+		Example: `# Update a Cluster Workflow Template:
+  argo cluster-template update FILE1
+	
+# Update a Cluster Workflow Template and print it as YAML:
+  argo cluster-template update FILE1 --output yaml
+  
+# Update a Cluster Workflow Template with relaxed validation:
+  argo cluster-template update FILE1 --strict false
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				cmd.HelpFunc()(cmd, args)
+				os.Exit(1)
+			}
+
+			updateClusterWorkflowTemplates(cmd.Context(), args, &cliUpdateOpts)
+		},
+	}
+	command.Flags().StringVarP(&cliUpdateOpts.output, "output", "o", "", "Output format. One of: name|json|yaml|wide")
+	command.Flags().BoolVar(&cliUpdateOpts.strict, "strict", true, "perform strict workflow validation")
+	return command
+}
+
+func updateClusterWorkflowTemplates(ctx context.Context, filePaths []string, cliOpts *cliUpdateOpts) {
+	if cliOpts == nil {
+		cliOpts = &cliUpdateOpts{}
+	}
+	ctx, apiClient := client.NewAPIClient(ctx)
+	serviceClient, err := apiClient.NewClusterWorkflowTemplateServiceClient()
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
this block in all of the functions I do think could be re-used in a helper as well (that's why I included these lines in my previous review [comments](https://github.com/argoproj/argo-workflows/pull/12803#discussion_r1536717614)), but I think the changes you have now are a good first pass and we can do more refactoring later. (like I think a huge chunk of the CLI could potentially be consolidated by passing resource specific functions around)

separating `generate*` from the networking portion as you did is also good practice.

Reply:
Yes, I thought about this reuse. However, considering that cliUpdateOpts and cliCreateOpts may be different, and in the current code
```
ctx, apiClient := client.NewAPIClient(ctx)
serviceClient = ··· 
```
These two lines are almost independent. Considering consistency, I did not encapsulate them into a function.

Reply:
Yea I think we could encapsulate to a CLI function that nearly every command could use (pseudo-code) `createClient(cliOpts, cliUpdateOpts, ClusterWorkflowTemplateServiceClient)`, but that would be a good bit more complicated, especially typing-wise (and perhaps not worth the complexity) 😅 Something to consider for a future refactor though.